user_config.c

```diff
@@ -0,0 +1,73 @@
+#include "user_config.h"
+#include <stdio.h>
+
+static uint8_t set_debug_level = 0, change_function = 0, temp_num = 0;
+static bool set_sensor_messages = false;
+static char save_type;
+
+// function takes in a message and parses it to see if max_debug_level() or allow_sensor_messages() needs to be changed
+void parse_usb_string(const char *input) {
+	uint8_t iterate = 0;
+	while (input[iterate] != '\0') {
+		switch (change_function) {
+			// Case 2: last check for if there has been a valid message sent
+			// --> if parse_usb_string has received a semicolon
+			case 2:
+				change_function = 0;
+				if (input[iterate] == ';') {
+					if (save_type == 'G')
+						set_debug_level = temp_num;
+					else if (save_type == 'S')
+						set_sensor_messages = temp_num;
+					else if (save_type == 'L') {
+						printf("Current Config: \n");
+						printf("\tMax debug level: %d\n", max_debug_level());
+						if (allow_sensor_messages())
+							printf("\tSensor messages are allowed!\n");
+						else
+							printf("\tSensor messages are not allowed!\n");
+					}
+					break;
+				} // @suppress("No break at end of case")
+
+
+			// Case 0: new instance of checking for valid message sent
+			// --> if parse_usb_string has received a char (either 'S' or 'G')
+			case 0:
+				if (input[iterate] == 'G') {
```

Comment:
This is probably a good time to use a switch statement. It'll save you a couple lines, and it'll be easier to add new commands in the future, if we find any.